lerobot/common/robot_devices/robots/manipulator.py

```diff
@@ -445,134 +556,126 @@ def teleop_step(
             before_camread_t = time.perf_counter()
             images[name] = self.cameras[name].async_read()
+            images[name] = torch.from_numpy(images[name])
             self.logs[f"read_camera_{name}_dt_s"] = self.cameras[name].logs["delta_timestamp_s"]
             self.logs[f"async_read_camera_{name}_dt_s"] = time.perf_counter() - before_camread_t
 
         # Populate output dictionnaries and format to pytorch
         obs_dict = {}
-        obs_dict["observation.state"] = torch.from_numpy(state)
+        obs_dict["observation.state"] = state
         for name in self.cameras:
-            obs_dict[f"observation.images.{name}"] = torch.from_numpy(images[name])
+            obs_dict[f"observation.images.{name}"] = images[name]
         return obs_dict
 
-    def send_action(self, action: torch.Tensor, follower_names: list[str] | None = None):
+    def send_action(self, action: torch.Tensor) -> torch.Tensor:
         """Command the follower arms to move to a target joint configuration.
 
         The relative action magnitude may be clipped depending on the configuration parameter
-        `max_relative_target`.
+        `max_relative_target`. In this case, the action sent differs from original action.
+        Thus, this function always returns the action actually sent.
 
         Args:
-            action: tensor containing the concatenated joint positions for the follower arms.
-            follower_names: Pass follower arm names to only control a subset of all the follower arms.
+            action: tensor containing the concatenated goal positions for the follower arms.
         """
         if not self.is_connected:
             raise RobotDeviceNotConnectedError(
-                "KochRobot is not connected. You need to run `robot.connect()`."
-            )
-
-        if follower_names is None:
-            follower_names = list(self.follower_arms)
-        elif not set(follower_names).issubset(self.follower_arms):
-            raise ValueError(
-                f"You provided {follower_names=} but only the following arms are registered: "
-                f"{list(self.follower_arms)}"
+                "ManipulatorRobot is not connected. You need to run `robot.connect()`."
             )
 
+        # Get goal position of each follower arm by splitting the action vector
         from_idx = 0
         to_idx = 0
         follower_goal_pos = {}
-        for name in follower_names:
-            to_idx += len(self.follower_arms[name].motor_names)
-            this_action = action[from_idx:to_idx]
+        for name in self.follower_arms:
+            if name in self.follower_arms:
+                to_idx += len(self.follower_arms[name].motor_names)
+                follower_goal_pos[name] = action[from_idx:to_idx]
+                from_idx = to_idx
+
+        # Send goal position to each follower
+        action_sent = []
+        for name in self.follower_arms:
+            goal_pos = follower_goal_pos[name]
```

Comment:
Nit but why are these in 2 separate loops? Makes it hard to read

Reply:
Removed the second for loop to simplify